Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/model/ChanThread.java

```diff
@@ -18,18 +18,24 @@
 
 import com.github.adamantcheese.chan.core.model.orm.Loadable;
 
-import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 
 public class ChanThread {
     private Loadable loadable;
     private List<Post> posts;
+    // Unmodifiable list of posts. We need it to make this class "thread-safe" (it's actually
```

Comment:
No. If you take a look at how UnmodifiableList is implemented you will see that the only thing it does is it stores internally a reference to the list you provide to it and uses it when you do operations like get/indexOf etc or throws an exception if you try to call any method that modifies a list (add/remove). So basically you don't need to update it. You just need to make sure that the original reference is not changed and if it is then you need to create a new unmodifiableList.

Reply:
>Is there any reason why we need posts to be thread-safe?

Haven't we had bugs because of posts because accessed from multiple threads? Or was it loadable? In any case, those posts at least are being accessed from multiple threads.

Reply:
Kinda garbage how that works around the whole "unmodifiable" principle. From what I can tell, posts itself can be unmodifiable and we won't need the extra read-only variable; the list is only ever read from, never written to.

Reply:
Well the whole point of unmodifiable collections is to not allow to add new items to it. it's more like "read-only" collection.

>From what I can tell, posts itself can be unmodifiable and we won't need the extra read-only variable

Actually yes. I can just make the "posts" unmodifiable collection and that's should be enough.

Reply:
Done.